src/Kunstmaan/MediaBundle/Helper/MediaManager.php

```diff
@@ -131,26 +186,30 @@ public function removeMedia(Media $media)
     /**
      * @param mixed $data
      *
-     * @return Media
+     * @return mixed
      */
     public function createNew($data)
     {
-        foreach ($this->handlers as $handler) {
-            $result = $handler->createNew($data);
-            if ($result) {
-                return $result;
+        if ($this->isExtensionAllowed($data)) {
+            foreach ($this->handlers as $handler) {
+                $result = $handler->createNew($data);
+                if ($result) {
+                    return $result;
+                }
             }
+
+            return null;
         }
 
-        return null;
+        return false;
```

Comment:
`false` is an unexpected return type compared to the previous code (kind of a bc break), would it make sense to return `null` when the extension is not allowed?